python/sparkdl/param/converters.py

```diff
@@ -0,0 +1,120 @@
+# Copyright 2017 Databricks, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import six
+
+import tensorflow as tf
+
+from pyspark.ml.param import TypeConverters
+
+import sparkdl.graph.utils as tfx
+import sparkdl.utils.keras_model as kmutil
+
+__all__ = ['SparkDLTypeConverters']
+
+def _get_strict_tensor_name(_maybe_tnsr_name):
+    assert isinstance(_maybe_tnsr_name, six.string_types), \
+        "must provide a strict tensor name as input, but got {}".format(type(_maybe_tnsr_name))
+    assert tfx.as_tensor_name(_maybe_tnsr_name) == _maybe_tnsr_name, \
+        "input {} must be a valid tensor name".format(_maybe_tnsr_name)
+    return _maybe_tnsr_name
+
+def _try_convert_tf_tensor_mapping(value, is_key_tf_tensor=True):
```

Comment:
do not use default arguments in private functions unless there are some very compelling reasons. This is a user-facing feature for API design.

Reply:
As a part of refactoring, this function has changed already.

In general, though, I would say having default arguments is quite useful at least for the following reasons. 
1. Serve as type annotation: in most IDEs, the signature is the only thing to show when browsing code. 
2. In certain cases, provide meaningful default settings.